### PR TITLE
updating docs for mbstyle.

### PR DESCRIPTION
**Pull request by vickdw**
@tbarsballe and @jodygarnett Please review.

### CODE REVIEW

File: docs/user/unsupported/mbstyle.rst [hunk 1]
```diff
@@ -3,6 +3,11 @@ Mapbox Styles Module
 
 The **gt-mbstyle** module is an unsupported module that provides a parser/encoder to convert between Mapbox Styles and GeoTools style objects. These docs are under active development, along with the module itself.
 
+* Human readable alternative to machine readable XML
```
Comment (jodygarnett) [inline docs/user/unsupported/mbstyle.rst]: Not technically true, json is a machine readable format.

File: docs/user/unsupported/mbstyle.rst [hunk 2]
```diff
@@ -3,6 +3,11 @@ Mapbox Styles Module
...
+* Compact representation
+* A JSON object with specific root level and nested properties
+* Full access to GeoTools rendering engine not limited by SLD specification
```
Comment (jodygarnett) [inline docs/user/unsupported/mbstyle.rst]: MBStyle access less of the rendering engine then SLD or YSLD.

If you like we can promote this as a positive, high level configuration of the geotools rendering engine, automatically applying appropriate "vendor options" for legibility (to match the intent of the mbstyle).

File: docs/user/unsupported/mbstyle.rst [hunk 3]
```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
...
+    List<MBLayer> layers = mbstyle.layers( "mapbox://mapbox.mapbox-streets-v6", "water" );
 
     //Which can be used to Generate Styler Layer Descriptor:
-    StyleLayerDescriptor sld = MBStyleTransformer.transform( layers );
\ No newline at end of file
+    StyleLayerDescriptor sld = MBStyleTransformer.transform( layers );
+
+MapBox Types
+^^^^^^^^^^^^
+ copied from the `MapBox Style Specification <https://www.mapbox.com/mapbox-gl-js/style-spec/>`_
```
Comment (tbarsballe) [inline docs/user/unsupported/mbstyle.rst]: What does this look like when compiled? I expect it should followed by a newline for rst formatting to work properly.

Also, should start with a capital letter (?)

File: docs/user/unsupported/mbstyle.rst [hunk 4]
```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
...
+  Color
+  '''''
+    Colors are written as JSON strings in a variety of permitted formats: HTML-style hex values, rgb, rgba, hsl, and hsla. Predefined HTML colors names, like yellow and blue, are also permitted.
```
Comment (tbarsballe) [inline docs/user/unsupported/mbstyle.rst]: Do we actually support alpha values in the MBStyle code right now?

File: docs/user/unsupported/mbstyle.rst [hunk 5]
```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
...
+  '''''''''''''''
+
+  .. code-block:: json
+
+    {
+      "version": 8, <Required>
+      "name": "Mapbox Streets",
+      "sprite": "mapbox://sprites/mapbox/streets-v8",
+      "glyphs": "mapbox://fonts/mapbox/{fontstack}/{range}.pbf",
+      "sources": {...}, <Required>
```
Comment (tbarsballe) [inline docs/user/unsupported/mbstyle.rst]: While `sources` is required by the spec, the current MBStyle geotools implementation does not enforce this. Maybe add a note to this effect?

File: docs/user/unsupported/mbstyle.rst [hunk 6]
```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
...
-  
+
     MBStyle mbstyle = MapBoxStyleParser.parse( reader );
 
     // pull back all layers for a provided source
     List<MBLayer> layers = mbstyle.layers( "mapbox://mapbox.mapbox-streets-v6" );
 
     // pull back selected layers for a provided source
-    List<MBLayer> layers = mbstyle.layers( "mapbox://mapbox.mapbox-streets-v6", "water" ); 
+    List<MBLayer> layers = mbstyle.layers( "mapbox://mapbox.mapbox-streets-v6", "water" );
```
Comment (jodygarnett) [inline docs/user/unsupported/mbstyle.rst]: we need a gwc example here

File: docs/user/unsupported/mbstyle.rst [hunk 7]
```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
...
 
     //Which can be used to Generate Styler Layer Descriptor:
-    StyleLayerDescriptor sld = MBStyleTransformer.transform( layers );
\ No newline at end of file
+    StyleLayerDescriptor sld = MBStyleTransformer.transform( layers );
+
+MapBox Types
+^^^^^^^^^^^^
+ copied from the `MapBox Style Specification <https://www.mapbox.com/mapbox-gl-js/style-spec/>`_
+  Color
```
Comment (jodygarnett) [inline docs/user/unsupported/mbstyle.rst]: Need a bit more than copied, see https://github.com/geotools/geotools/pull/1576 for example text

File: docs/user/unsupported/mbstyle.rst [hunk 8]
```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
...
+            "text-opacity": "", <Number, defaults to 1 requires text-field>
+            "text-halo-color": "", <Color, defaults to rgba(0,0,0,0) requires text-field>
+            "text-halo-width": "", <Number, units in pixels, defaults to 0 requires text-field>
+            "text-halo-blur": "", <Number, units in pixels, defaults to 0 requires text-field>
+            "text-translate": "", <Array units in pixels defaults to [0,0] requires text-field>
+            "text-translate-anchor": "" <Enum, [map, viewport] defaults to map, requires text-field, text-translate>
+          }
+        }
+      ],
+    }
```
Comment (tbarsballe) [inline docs/user/unsupported/mbstyle.rst]: Add a newline after this?

File: docs/user/unsupported/mbstyle.rst [hunk 9]
```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
...
+        {
+          "line-color": "#ff0",
+          "line-color": "#ffff00",
+          "line-color": "rgb(255, 255, 0)",
+          "line-color": "rgba(255, 255, 0, 1)",
+          "line-color": "hsl(100, 50%, 50%)",
+          "line-color": "hsla(100, 50%, 50%, 1)",
+          "line-color": "yellow"
+        }
+    Especially of note is the support for hsl, which can be easier to reason about than rgb().
```
Comment (jodygarnett) [inline docs/user/unsupported/mbstyle.rst]: Although the mbstyle spec has this approach, however I prefer working cut and paste examples if we can do it.

File: docs/user/unsupported/mbstyle.rst [hunk 10]
```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
...
+      }
+
+  String
+  ''''''
+    A string is basically just text. In Mapbox styles, you're going to put it in quotes. Strings can be anything, though pay attention to the case of text-field - it actually will refer to features, which you refer to by putting them in curly braces, as seen in the example below.
+
+    .. code-block:: json
+
+      {
+        "text-field": "{MY_FIELD}"
```
Comment (jodygarnett) [inline docs/user/unsupported/mbstyle.rst]: We list a string, but we show a field, need two examples here.

Note: Why the difference? You can also have an identity function to generate an enum from a field so we need to keep these examples easy to follow.

File: docs/user/unsupported/mbstyle.rst [hunk 11]
```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
...
+    Use the HCL color space to interpolate color values, interpolating the Hue, Chroma, and Luminance channels individually.
+
+    **Zoom Functions** allow the appearance of a map feature to change with map’s zoom level. Zoom functions can be used to create the illusion of depth and control data density. Each stop is an array with two elements: the first is a zoom level and the second is a function output value.
+
+    .. code-block:: json
+
+      {
+        "circle-radius": {
+          "stops": [
+            // zoom is 5 -> circle radius will be 1px
```
Comment (jodygarnett) [inline docs/user/unsupported/mbstyle.rst]: I am not sure if we support comments, if we do not support comments then we should not use it in examples.

File: docs/user/unsupported/mbstyle.rst [hunk 12]
```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
...
+          },
+          "paint": {
+            "background-opacity": 0.45,
+            "background-color": "#00FF00"
+          }
+        }
+      ]
+    }
+
+  *background-color* is disabled by the presence of *background-pattern*
```
Comment (jodygarnett) [inline docs/user/unsupported/mbstyle.rst]: We need a compatibility table here that says we do not support background layer yet.

File: docs/user/unsupported/mbstyle.rst [hunk 13]
```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
...
+            "fill-color": "#FF595E",
+            "fill-outline-color":"#1982C4",
+            "fill-translate": [20,20],
+            "fill-translate-anchor": "map",
+            "fill-pattern": <String>
+          }
+        }
+      ]
+    }
+
```
Comment (jodygarnett) [inline docs/user/unsupported/mbstyle.rst]: These are nice standalone examples. fill-pattern <String> looks odd?

File: docs/user/unsupported/mbstyle.rst [hunk 14]
```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
...
+      "layers": [
+        {
+          "id": "test-id",
+          "type": "line",
+          "source": "test-source",
+          "source-layer": "test-source-layer",
+          "layout": {
+              "line-cap": "square",
+              "line-join": "round",
+              "line-mitre-limit": 2, <Optional - Requires line-join=mitre>
```
Comment (jodygarnett) [inline docs/user/unsupported/mbstyle.rst]: we have a different kind of comment here, please remove or confirm if json supports comments.

File: docs/user/unsupported/mbstyle.rst [hunk 15]
```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
...
+
+    {
+      "layers": [
+        {
+          "id": "test-id",
+          "type": "symbol",
+          "source": "test-source",
+          "source-layer": "test-source-layer",
+          "layout": {
+              "symbol-placement": "", <Enum, [Point, line] Defaults to Point>
```
Comment (jodygarnett) [inline docs/user/unsupported/mbstyle.rst]: okay so this is more of a grammer and less of an example?

**Comment by tbarsballe** [comment]
Travis CI fails, looks like an RST error:

```
     [exec] Warning, treated as error:
     [exec] /home/travis/build/geotools/geotools/docs/user/unsupported/mbstyle.rst:43: SEVERE: Unexpected section title.

     [exec] Color
     [exec] '''''
```

**Comment by tbarsballe** [comment]
Is this still relevant, or has it been superceeded by https://github.com/geotools/geotools/pull/1584 ?

**Comment by bencaradocdavies** [comment]
This PR breaks the docs build: 
'''
     [exec] Warning, treated as error:
     [exec] /home/travis/build/geotools/geotools/docs/user/unsupported/mbstyle.rst:43: SEVERE: Unexpected section title.

     [exec] Color
     [exec] '''''
'''

**Pull request merged**